lib/ansible/modules/files/lineinfile.py

```diff
@@ -188,8 +188,8 @@
 - name: Ensure the JBoss memory settings are exactly as needed
   lineinfile:
     path: /opt/jboss-as/bin/standalone.conf
-    regexp: '^(.*)Xms(\\d+)m(.*)$'
-    line: '\1Xms${xms}m\3'
+    regexp: '^(.*)Xms(\d+)m(.*)$'
+    line: '\1Xms{{ xms }}m\3'
```

Comment:
@grimsy thanks for reviewing the Ansible docs. From @samdoran's comment it seems that the change suggested here isn't quite right. But your experience suggests that the docs could use clarification. 

Have you made the example work for your use case? Do you have suggestions for updating the docs to make them easier to follow for the next person? Feel free to post them in a comment if you can't update the branch for this PR.

If we don't hear back in a couple of weeks we'll close this PR. In the meantime I'm marking it so we don't merge the existing change by mistake.

Reply:
Hi @acozine,
thanks for the follow up - I'm actually using it to set the Xmx/Xms in Tomcat rather than JBoss, so there could be differences in the format of the file which I didn't take into account.
The first part was the NOTE on the example:
# NOTE: Yaml requires escaping backslashes in double quotes but not in single quotes
yet the example had escaped backslashes in single quotes.
The second part was the variable substitution - I'm using it to write the variable value hence using jinja variable syntax, I'm not sure if JBoss literally wants the shell variable placeholder so that may not be valid in my PR.

Reply:
I would agree with the change on line 191: there is no need to escape the backslash in YAML when the line is quoted with single quotes.

The change in line 192 is incorrect and should be removed from this PR.